device/coordinate_manager.cpp

```diff
@@ -38,179 +38,53 @@ CoordinateManager::CoordinateManager(
-            return dram_logical_to_physical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_logical_to_physical;
-        case CoreType::ARC:
-            return arc_logical_to_physical;
-        case CoreType::PCIE:
-            return pcie_logical_to_physical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting logical to physical mapping");
-    }
-}
-
-std::map<tt_xy_pair, CoreCoord>& CoordinateManager::get_physical_to_logical(CoreType core_type) {
-    switch (core_type) {
-        case CoreType::TENSIX:
-            return tensix_physical_to_logical;
-        case CoreType::DRAM:
-            return dram_physical_to_logical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_physical_to_logical;
-        case CoreType::ARC:
-            return arc_physical_to_logical;
-        case CoreType::PCIE:
-            return pcie_physical_to_logical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting physical to logical mapping");
-    }
-}
-
-std::map<tt_xy_pair, CoreCoord>& CoordinateManager::get_virtual_to_logical(CoreType core_type) {
-    switch (core_type) {
-        case CoreType::TENSIX:
-            return tensix_virtual_to_logical;
-        case CoreType::DRAM:
-            return dram_virtual_to_logical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_virtual_to_logical;
-        case CoreType::ARC:
-            return arc_virtual_to_logical;
-        case CoreType::PCIE:
-            return pcie_virtual_to_logical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting virtual to logical mapping");
-    }
-}
-
-std::map<tt_xy_pair, CoreCoord>& CoordinateManager::get_translated_to_logical(CoreType core_type) {
-    switch (core_type) {
-        case CoreType::TENSIX:
-            return tensix_translated_to_logical;
-        case CoreType::DRAM:
-            return dram_translated_to_logical;
-        case CoreType::ACTIVE_ETH:
-        case CoreType::IDLE_ETH:
-        case CoreType::ETH:
-            return eth_translated_to_logical;
-        case CoreType::ARC:
-            return arc_translated_to_logical;
-        case CoreType::PCIE:
-            return pcie_translated_to_logical;
-        default:
-            throw std::runtime_error("Core type is not supported for getting translated to logical mapping");
-    }
-}
-
 CoreCoord CoordinateManager::to_physical(const CoreCoord core_coord) {
     switch (core_coord.coord_system) {
         case CoordSystem::PHYSICAL:
             return core_coord;
         case CoordSystem::VIRTUAL:
         case CoordSystem::TRANSLATED:
-            return to_physical(to_logical(core_coord));
+        case CoordSystem::LOGICAL: {
```

Comment:
I recommend adding a default case to the switch statements to throw an error otherwise if a new enum value is added it could become very hard to remember and find where it's missing

Reply:
removed to_physical, everything goes through to function now